lib/omniauth/strategies/slack.rb

```diff
@@ -19,56 +19,37 @@ class Slack < OmniAuth::Strategies::OAuth2
         param_name: 'token'
       }
 
-      uid { raw_info['user_id'] }
+      # User ID is not guaranteed to be globally unique across all Slack users.
+      # The combination of user ID and team ID, on the other hand, is guaranteed
+      # to be globally unique.
+      uid { "#{user_identity['id']}-#{team_identity['id']}" }
 
       info do
         hash = {
-          nickname: raw_info['user'],
-          team: raw_info['team'],
-          user: raw_info['user'],
-          team_id: raw_info['team_id'],
-          user_id: raw_info['user_id']
```

Comment:
What was the resolution to this @z-productions?

I'm using the sign-in flow, and I'm considering updating from 2.3.0 to this latest code in order to better support the sign-in flow. But what I need is to get the `team_id` when someone goes through the sign-in flow, and as far as I can tell the only way that I can get the `team_id` using this strategy is by parsing it out of the `uid`, which seems less than ideal. (I can't get it from `raw_info` because I'm only requesting `identity.basic` for sign-in, so I can't rely on `team:read`.)